extras/ai-battle/main.cpp

```diff
@@ -0,0 +1,127 @@
+// Copyright (C) 2005 - 2024 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "GlobalGameSettings.h"
+#include "HeadlessGame.h"
+#include "QuickStartGame.h"
+#include "RTTR_Version.h"
+#include "RttrConfig.h"
+#include "files.h"
+#include "random/Random.h"
+#include "s25util/System.h"
+
+#include <boost/filesystem.hpp>
+#include <boost/nowide/args.hpp>
+#include <boost/nowide/filesystem.hpp>
+#include <boost/nowide/iostream.hpp>
+#include <boost/optional.hpp>
+#include <boost/program_options.hpp>
+
+namespace bnw = boost::nowide;
+namespace bfs = boost::filesystem;
+namespace po = boost::program_options;
+
+int main(int argc, char** argv)
+{
+    bnw::nowide_filesystem();
+    bnw::args _(argc, argv);
+
+    boost::optional<std::string> replay_path;
+    boost::optional<std::string> savegame_path;
+
+    po::options_description desc("Allowed options");
+    // clang-format off
+    desc.add_options()
+        ("help,h", "Show help")
+        ("map,m", po::value<std::string>()->required(),"Map to load")
+        ("ai", po::value<std::vector<std::string>>()->required(),"AI player(s) to add")
+        ("objective", po::value<std::string>()->default_value("domination"),"domination(default)|conquer")
+        ("replay", po::value(&replay_path),"Filename to write replay to (optional)")
+        ("save", po::value(&savegame_path),"Filename to write savegame to (optional)")
+        ("random_init", po::value<unsigned>()->default_value(std::chrono::high_resolution_clock::now().time_since_epoch().count()),"Seed value for the random number generator (optional)")
```

Comment:
Compiler complains about signed->unsigned.
1. use an optional<unsigned> and `value_or(std::chrono...)`
2. Or: `const auto defaultSeed = static_cast<unsigned>(std::chrono::high_resolution_clock::now().time_since_epoch().count())` next to replay/savegame_path and use it here